octodns/provider/yaml.py

```diff
@@ -162,6 +218,69 @@ def supports(self, record):
     def SUPPORTS_ROOT_NS(self):
         return self.supports_root_ns
 
+    def list_zones(self):
+        self.log.debug('list_zones:')
+        zones = set()
+
+        extension = self.split_extension
+        if extension:
+            # we want to leave the .
+            trim = len(extension) - 1
+            self.log.debug(
+                'list_zones:   looking for split zones, trim=%d', trim
+            )
+            for dirname in listdir(self.directory):
+                not_ends_with = not dirname.endswith(extension)
+                not_dir = not isdir(join(self.directory, dirname))
+                if not_dir or not_ends_with:
+                    continue
+                if trim:
+                    dirname = dirname[:-trim]
+                zones.add(dirname)
+
+        if not self.split_extension:
+            self.log.debug('list_zones:   looking for zone files')
```

Comment:
Does this mean that a single instance of YamlProvider can't be used to support a hybrid of non-split and split yaml files? Looks like if I've set `split_extension` to non-empty/non-null, it will skip listing non-split zones altogether.